PyInstaller/loader/pyi_archive.py

```diff
@@ -121,7 +158,7 @@ def loadtoc(self):
         self.lib.seek(self.start + self.TOCPOS)
         (offset,) = struct.unpack('!i', self.lib.read(4))
         self.lib.seek(self.start + offset)
-        self.toc = marshal.load(self.lib)
+        self.toc = marshal.loads(self.lib.read())
```

Comment:
I'm curious: ArchiveFile passes all getattr-calls onto the file-handler, so why do we have to explicitly use `read()` here?